src/uct/cuda/cuda_copy/cuda_copy_md.c

```diff
@@ -119,6 +119,31 @@ UCS_PROFILE_FUNC(ucs_status_t, uct_cuda_copy_mem_dereg,
     return UCS_OK;
 }
 
+static ucs_status_t uct_cuda_copy_mem_alloc(uct_md_h md,
+                                            size_t *length_p,
+                                            void **address_p,
+                                            unsigned flags,
```

Comment:
The new API in #5562 introduces mem_type parameter here. When that goes in, cuda_copy should check that and allocate pinned device memory if mem_type is CUDA or pinned host memory (through [cumemallochost](https://docs.nvidia.com/cuda/cuda-driver-api/group__CUDA__MEM.html#group__CUDA__MEM_1gdd8311286d2c2691605362c689bc64e0)) if mem_type is HOST

Reply:
@ssaulters the changes are in master now. Can you rebase and alter the function?